tensorflow/models/image/mnist/convolutional.py

```diff
@@ -296,11 +296,13 @@ def eval_in_batches(data, sess):
       # node in the graph it should be fed to.
       feed_dict = {train_data_node: batch_data,
                    train_labels_node: batch_labels}
-      # Run the graph and fetch some of the nodes.
-      _, l, lr, predictions = sess.run(
-          [optimizer, loss, learning_rate, train_prediction],
-          feed_dict=feed_dict)
+      # Run the optimizer to update weights.
+      sess.run(optimizer, feed_dict = feed_dict)
```

Comment:
I am using i76900k and GTX980. The major improvement is from [_train_prediction_,](https://github.com/willdzeng/tensorflow/blob/6063e6062d920cb4b666c29b5c49f37521cf3edc/tensorflow/models/image/mnist/convolutional.py#L255) which is not calculated every time in _optimizer_.
It's around 10% improvement from 110s to 95s for me.